flake.nix

```diff
@@ -60,13 +60,15 @@
           fmt = flake-utils.lib.mkApp {
             drv = with pkgs;
               writeScriptBin "fmt" ''
+                #!/bin/bash
```

Comment:
I would remove that as the assumption of `/bin/bash` is not guaranteed. I think only shell is always available.
One should replace the `writeScriptBin` with `writeShellScriptBin`. Accordingly the `#!/bin/bash` is not required and the correct shell runtime is populated.

https://nixos.org/manual/nixpkgs/unstable/#trivial-builder-writeShellScriptBin